fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProviderManager.java

```diff
@@ -77,6 +77,22 @@ private static String getProviderPath(String providerName) {
                 "%s/dimensionCache/%s/lucene_indexes/",
                 SYSTEM_CONFIG.getStringProperty(LUCENE_INDEX_PATH),
                 providerName
-        );
+        ).replaceAll("/+", "/"); // replaces one or more slashes with one slash:
+    }
+
+    /**
+     * Get the path for the lucene index files that are going to be deleted when we are cleaning up search provider.
+     *
+     * @param providersPath  Location of all providers
+     *
+     * @return the path to the lucene index files of all providers.
+     */
+    private static String getDeleteProviderPath(String providersPath) {
```

Comment:
Why do we need this and not use the old one?

Reply:
Because groovy test creates new directory called `new_lucene_indexes`, this cannot be deleted using the old one during test cleanup